src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala

```diff
@@ -252,11 +252,9 @@ private[health] class HealthCheckActor(
     // dead task (in addition to the living one), thus sometimes leading
     // Marathon to report the task / instance / app as unhealthy while
     // everything is running correctly.
-    val healthOfInstanceId = healthByTaskId.find(_._1.instanceId == instanceId)
-    if (healthOfInstanceId.isDefined)
-      healthByTaskId.remove(healthOfInstanceId.get._1)
+    healthIndex.remove(taskId)
```

Comment:
⚠️ the previous code ensured that the entry related to taskId wasn't removed if instance id didn't match, and here we're breaking that.